shared/chat/conversation-list/conversation-list-container.js

```diff
@@ -1,15 +1,133 @@
 // @flow
 import {namedConnect} from '../../util/container'
-import ConversationList from './conversation-list'
+import {memoize} from '../../util/memoize'
+import * as Types from '../../constants/types/chat2'
+import * as Constants from '../../constants/chat2'
+import * as Flow from '../../util/flow'
+import {isMobile} from '../../constants/platform'
+import ConversationList, {type SmallTeamRowItem, type BigTeamChannelRowItem} from './conversation-list'
+import getFilteredRowsAndMetadata from '../inbox/container/filtered'
 
 type OwnProps = {|
-  onSelect?: () => void,
+  filter?: string,
+  onSelect: (conversationIDKey: Types.ConversationIDKey) => void,
+  onSetFilter?: (filter: string) => void,
+  selected: Types.ConversationIDKey, // mobile only
 |}
 
-const mapStateToProps = state => ({})
+const notificationsTypeToNumber = (t: Types.NotificationsType): number => {
+  switch (t) {
+    case 'onAnyActivity':
+      return 1
+    case 'onWhenAtMentioned':
+      return 2
+    case 'never':
+      return 3
+    default:
+      Flow.ifFlowComplainsAboutThisFunctionYouHaventHandledAllCasesInASwitch(t)
+      return 0
+  }
+}
+
+const boolToNumber = (b: boolean): number => (b ? 1 : 0)
+
+const staleToNumber = (convTime: number, staleCutoff: number) => (convTime < staleCutoff ? 1 : 0)
+
+const getAWeekAgo = () => {
+  let t = new Date()
+  return t.setDate(t.getDate() - 7) // works fine for cross-boundary; returns a number
+}
+
+const getSortedConversationIDKeys = memoize(
```

Comment:
really all this should move to core, and will soon

Reply:
Cool; keep this for now? Also note that this is different from inbox sorting.